FetchDependency.cmake

```diff
@@ -343,6 +332,8 @@ function(fetch_dependency FD_NAME)
       endforeach()
     endif()
 
+    # Write the options file now that we know it has succeeded in configuring.
+    file(WRITE ${OptionsFilePath} "${RequiredOptions}\n")
```

Comment:
I think this is fine for now, but not deleting or otherwise invalidating something when we start to build means we can get into a state where, for example, pulling a bad tag or doing a local edit gets you into a place where we won't rebuild after the initial failure. 

A local change I'm working on addresses that as a side-effect, but I should maybe do some profiling and seeing if trying to manually skip the build is really worth it (like can I just rely on the build system's no-op check, the cost to actually invoke the `cmake` process to run that check is pretty low).